### DISCUSSION TITLE
How to change the balancing payload depending on http status code?

### DISCUSSION THREAD

**Discussion by milapointe** [ThreeMammals/Ocelot#2140] [Q&A] [answered]
Hi,

I have a use case that I have 2 downstream servers.

Depending on the response on the first server, I would like to either :

- Call the second downstream server only on http code 429
or
- Change the payload and http code so that I can add a set location 302 to another api

Is it something supported?

Thanks

**Comment by raman-m** [comment, marked as answer]
Hello!
Welcome to Ocelot world! 🐯 

> I have a use case that I have 2 downstream servers.

It appears that your production environment is relatively small, considering you have a use case involving only two downstream servers. 😃 

> Depending on the response on the first server, I would like to either :
> 
> - Call the second downstream server only on http code 429
> or
> - Change the payload and http code so that I can add a set location 302 to another api
> 
> Is it something supported?
> 

#### No❕ 
Currently, there is no official functionality for such a scenario. However, you can create a custom solution. You would need to develop a [Custom Load Balancer](https://ocelot.readthedocs.io/en/latest/features/loadbalancer.html#custom-load-balancers) that will select the second server based on the response from the first one. Additionally, you'll need to implement a special [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) to capture the response status and store it in `HttpContext.Items` for the [custom load balancer](https://ocelot.readthedocs.io/en/latest/features/loadbalancer.html#custom-load-balancers), which will then provide the second service instance based on the trigger and predicate from the custom delegating handler.

Should you decide to proceed with development and require further consultation, please contact us to discuss a development roadmap.

**P.S.** Please inform us if there are other gateways with similar functionality.

**Comment by milapointe** [comment]
Hi, thanks for the anwser. 

About the size, I said 2 downstream servers but in reality, it's more an container with ocelot loaded as a sidecar that will call an API outside to endpoint1 and if this one return a http code different from 200, we call endpoint2 or replace payload with 302 to call that other endpoint2.

About others gateway that could support that use case, Broadcom/VMware (formely Layer7) do that kind of use case but invole much more stuff that just a sidecar container.

Thanks

**Reply by raman-m** [reply]
Typically HTTP clients should decide how to react to non-2xx statuses, but this would be client-side logic.
If the developer needs gateway-side logic, the recipe is above.
In theory, such logic could be converted into a new Ocelot's feature, but the possible design is not clear to me yet.

**Reply by raman-m** [reply]
> About others gateway that could support that use case, Broadcom/VMware (formely Layer7) do that kind of use case but invole much more stuff that just a sidecar container.

No, I meant software gateways like Ocelot, not hardware ones.
Current popular software gateways are [Microsoft Reverse Proxy](https://github.com/microsoft/reverse-proxy), [Envoy Proxy](https://github.com/envoyproxy/envoy), [Kong API Gateway](https://github.com/Kong/kong), [Traefik Proxy](https://github.com/traefik/traefik), Kubernetes Ingress, Nginx etc.
